src/Aspire.Hosting/Dashboard/DashboardConsoleLogCollector.cs

```diff
@@ -0,0 +1,174 @@
+                    }
+                    break;
+                }
+
+                var subscriptionTask = Task.Run(async () =>
+                {
+                    await foreach (var batch in resourceSubscription.Subscription.WithCancellation(cancellationToken))
+                    {
+                        foreach (var change in batch)
+                        {
+                            if (change.Resource.DisplayName == KnownResourceNames.AspireDashboard)
+                            {
+                                if (change.ChangeType == ResourceSnapshotChangeType.Upsert && change.Resource.State == ExecutableState.Running)
+                                {
+                                    ProcessDashboardResource(change.Resource);
+                                }
+                                else
+                                {
+                                    lock (_lock)
+                                    {
+                                        _dashboardResource = null;
+                                        _resourceSubscriptionCts?.Cancel();
+                                        _resourceSubscriptionCts = null;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }, cancellationToken);
+
+                await subscriptionTask.ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogDebug(ex, "Error collecting dashboard logs. Retrying.");
+                await Task.Delay(TimeSpan.FromSeconds(1), cancellationToken).ConfigureAwait(false);
+            }
+        }
+    }
+
+    private async Task WatchLogs(ResourceSnapshot resourceSnapshot, CancellationToken cancellationToken)
+    {
+        try
+        {
+            var consoleLogsSubscription = _dashboardServiceData.SubscribeConsoleLogs(resourceSnapshot.Name);
+            if (consoleLogsSubscription == null)
+            {
+                throw new InvalidOperationException($"No subscription console logs returned for '{resourceSnapshot.Name}'.");
+            }
+
+            await foreach (var item in consoleLogsSubscription.WithCancellation(cancellationToken))
+            {
+                foreach (var log in item)
+                {
+                    var match = LogParsingConstants.Rfc3339RegEx.Match(log.Content);
+                    var resolvedContent = match.Success ? log.Content.Substring(match.Length + 1) : log.Content;
+
+                    if (!log.IsErrorMessage)
+                    {
+                        Console.Write("(Dashboard) ");
+                        Console.WriteLine(resolvedContent);
+                    }
+                    else
+                    {
+                        Console.Error.Write("(Dashboard) ");
+                        Console.Error.WriteLine(resolvedContent);
+                    }
+                }
```

Comment:
One big problem with this appraoch is overlapping with any other log statements that could happen concurrently.

Reply:
Yes. Logs from the dashboard are prefixed with `(Dashboard)` to make it clear where they're coming from.

Reply:
I was thinking he meant something else could sneak in between the Write and the WriteLine calls.

Reply:
I see. Combining them and then writing them will fix that.

Reply:
These should all just use ILogger so its syncrhronized.

Reply:
The problem with doing that is we're receiving raw console output from the dashboard app. Writing that to ILogger would result in that info being duplicated or wrong. We'd need to parse out the level, category, message, and put it back into a local ILogger.

That didn't seem worth it to me so I just wrote raw console output from the dashboard out to the console in the apphost.